src/sut/dumblog/app/Main.hs

```diff
@@ -1,17 +1,62 @@
 module Main where
 
 import Control.Concurrent.Async (async)
+import Control.Monad (when)
+import Journal (Journal)
 import qualified Journal
 import qualified Journal.Types.AtomicCounter as AtomicCounter
 import qualified Journal.Internal.Metrics as Metrics
 import Journal.Internal.Logger as Logger
+import System.Directory (doesFileExist)
 
 import Blocker (emptyBlocker)
+import Codec (Envelope(..), decode)
 import FrontEnd (runFrontEnd, FrontEndInfo(..))
 import Metrics (dumblogSchema)
-import StateMachine(initState)
+import StateMachine(InMemoryDumblog, initState, runCommand)
+import Types (Command)
 import Worker (worker, WorkerInfo(..))
 
+fetchJournal :: FilePath -> Journal.Options -> IO Journal
+fetchJournal fpj opts = do
+  firstTime <- doesFileExist fpj -- We should have some more sanity check?
```

Comment:
This is already done in the `allocateJournal`:

```haskell
  -- XXX: only for debugging:
  logg logger ("removing " ++ fp)
  removeFile fp
  b <- doesFileExist fp
  when (not b) $ do
    logg logger ("allocateJournal, creating new journal: " ++ fp)
   ...
```
We should remove the `removeFile` (I added that when I was changing the journal format and wanted a new journal to always be created, otherwise I would get bugs related to the old format).

Reply:
Okay I remove the `removeFile`.